File: jd_sellergoods_web/src/main/java/com/jd/sellergoods/controller/BrandController.java
package com.jd.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.jd.common.pojo.JdResult;
import com.jd.common.pojo.PageResult;
import com.jd.pojo.TbBrand;
import com.jd.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("brand")
public class BrandController {
   @Reference
   private BrandService brandService;

   @RequestMapping("show")
   public List<TbBrand> getAllBrand(){
      List<TbBrand> allBrand = brandService.getAllBrand();
      return allBrand;
   }

   /**
    * 分页查询 （angularjs设计好了 当前页面就叫page,每页显示的信息条数就叫rows）
    * @param pageNum
    * @param pageSize
    * @return
    */
   @RequestMapping("findPage")
   public PageResult findPage(@RequestParam(value = "pageNum",defaultValue = "1")Integer pageNum,
                              @RequestParam(value = "pageSize",defaultValue = "10")Integer pageSize){
      return brandService.findPage(pageNum,pageSize);
   }

   @RequestMapping("add")
   public JdResult add(@RequestBody TbBrand tbBrand){
      try {
         brandService.addBrand(tbBrand);
         return JdResult.ok();
      }catch (Exception e){
         e.printStackTrace();
         return new JdResult(false,"添加失败",null);
      }
   }

   /**
    * 根据id获取实体对象
    * @param id
    * @return
    */
   @RequestMapping("findOne")
   public TbBrand findOne(@RequestParam("id")Long id){
      return brandService.findOne(id);
   }

   @RequestMapping("update")
   public JdResult update(@RequestBody TbBrand tbBrand){
      try {
         brandService.updateBrand(tbBrand);
         return JdResult.ok();
      }catch (Exception e){
         return new JdResult(false,"修改失败-名称已存在",null);
      }
   }

   @RequestMapping("delete")
   public JdResult delete(@RequestParam("ids")Long[] ids){
      try {
         brandService.deleteById(ids);
         return JdResult.ok();
      }catch (Exception e){
         return new JdResult(false,"删除失败",null);
      }
   }

   @RequestMapping("search")
   public PageResult search(@RequestParam(value = "pageNum",defaultValue = "1")Integer pageNum,
                            @RequestParam(value = "pageSize",defaultValue = "10")Integer pageSize,
                            @RequestBody TbBrand tbBrand){
      return brandService.findPage(tbBrand,pageNum,pageSize);
   }
}

File: jd_sellergoods_web/src/main/webapp/js/controller/brandController.js
//自定义控制器
app.controller("brandController",function ($scope,$controller,brandService) {
    //控制器继承 引入baseController
    //$controller是angularJs的内置服务，通过这个内置服务可以实现伪继承--实际上就是与baseController共享$scope
    $controller("baseController",{$scope:$scope});

    //读取数据库表绑定到表单中
    $scope.findAll = function () {
        brandService.findAll().success(function (response) {
            $scope.list = response;
        });
    }


    $scope.findPage = function (pageNum,pageSize) {
        brandService.findPage(pageNum,pageSize).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems=response.total;
        })
    }

    //保存
    $scope.save = function () {
        var object = null;
        if ($scope.entity.id != null){
            object = brandService.update($scope.entity);
        }else {
            object = brandService.add($scope.entity);
        }

        object.success(function (response) {
            if (response.success){
                $scope.reloadList();
            }else {
                alert(response.message);
            }
        });
    }

    //修改
    $scope.findOne  = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        })
    }

    //删除
    $scope.delete = function () {
        brandService.delete($scope.selectIds).success(function (response) {
            if (response.success){
                $scope.reloadList();
                $scope.selectIds = [];
                alert(response.message);
            }else {
                alert(response.message);
            }
        })
    }

    //条件查询的对象
    $scope.searchEntity = {};
    //条件查询
    $scope.search = function (pageNum,pageSize) {
        brandService.search(pageNum,pageSize,$scope.searchEntity).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        });
    }
});